EmpowerPlant/EmpowerPlantViewController.swift

```diff
@@ -8,21 +8,149 @@
+        
         configureNavigationItems()
+
+        // TODO - clear Db on startup, and insert the objects from server into the db
+        getAllProductsFromDb()
+        getAllProductsFromServer()
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return products.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let model = products[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+        cell.textLabel?.text = model.title
+        return cell
+    }
+    
+    func getAllProductsFromServer() {
+        print("getAllProductsFromServer")
+        let url = URL(string: "https://application-monitoring-flask-dot-sales-engineering-sf.appspot.com/products-join")!
+        var request = URLRequest(url: url)
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        struct Product1: Decodable {
+
+            // MARK: - Properties
+            let id: Int
+            let title: String
+            let description: String
+            let descriptionfull: String
+            let img: String
+            let imgcropped: String
+            let price: Int
+            // reviews: [{id: 4, productid: 4, rating: 4, customerid: null, description: null, created: String},...]
+
+        }
+        
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
+            if let data = data {
+                // Can't use 'Product' here because it can't be Decoded, so testing with Product1 for now
+                if let products = try? JSONDecoder().decode([Product1].self, from: data) {
+                    // Prints the 4 products from server, successfully
+                    for product in products {
+                        print(product.title, product.description)
+                    }
+                } else {
+                    print("Invalid Response")
+                }
+            } else if let error = error {
+                print("HTTP Request Failed \(error)")
+            }
+        }
+        // print("Dev Note - I am printed before the products info above is printed")
+        task.resume()
+        
+        // Don't Reload Table, because we still have to conver the above JSON objects from the server into Swift objects
+//        do {
+//            DispatchQueue.main.async {
+//                self.tableView.reloadData()
+//            }
+//        }
+//        catch {
+//            //error
+//        }
+    }
+    
+    func getAllProductsFromDb() {
+        do {
+            self.products = try context.fetch(Product.fetchRequest())
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+        catch {
+            // error
+        }
+    }
+    
+    func createProduct(title: String) {
```

Comment:
Create/Delete/Update are just for testing the db and will be deprecated soon. No need for them.

Reply:
understood 👍